JS/index.js

```diff
@@ -0,0 +1,184 @@
+console.log('hello');
+let countries = [];
+const containerCountries = document.querySelector(".countries-grid");
+const countryCardTemplate = document.querySelector(".country.scale-effect");
+
+fetch('./CountriesData.json')
+    .then((res) => {
+        if (!res.ok) {
+            throw new Error(`HTTP error! Status: ${res.status}`);
+        }
+        return res.json();
+    })
+    .then((data) => {
+        countries = data;
+        console.log(data);
+        innerDetailsIntoCards(countries); 
+    })
+    .catch((error) => {
+        console.error("Unable to fetch data:", error);
+    });
+
+const innerDetailsIntoCards = (array) => {
+    containerCountries.innerHTML = '';
+
+    array.forEach((country) => {
+        const cloneCard = countryCardTemplate.cloneNode(true);
+       
+        const titleElement = cloneCard.querySelector(".country-title");
+        const flagImage = cloneCard.querySelector(".country-flag img");
+        const countryInfo = cloneCard.querySelectorAll(".country-brief li");
+
+        titleElement.innerHTML = country.name;
+        flagImage.src = country.flag;
+        flagImage.alt = `${country.name}`;
+        
+        countryInfo[0].innerHTML = `<strong>Population:</strong> ${country.population}`;
+        countryInfo[1].innerHTML = `<strong>Region:</strong> ${country.region}`;
+        countryInfo[2].innerHTML = `<strong>Capital:</strong> ${country.capital}`;
```

Comment:
The usage of `innerHTML` is not recommended due to security vulnerabilities.

https://medium.com/@jasen.miyamoto/appending-to-the-dom-and-the-dangers-of-using-innerhtml-debd77e53e70